lib/atlas/edge.rb

```diff
@@ -16,13 +16,14 @@ class Edge
 
     DIRECTORY = 'edges'
 
-    attribute :type,         Symbol
-    attribute :demand,       Float
-    attribute :parent_share, Float
-    attribute :child_share,  Float
-    attribute :reversed,     Boolean, default: false
-    attribute :priority,     Integer
-    attribute :groups,       Array[Symbol]
+    attribute :type,          Symbol
+    attribute :demand,        Float
+    attribute :parent_share,  Float
+    attribute :child_share,   Float
+    attribute :reversed,      Boolean, default: false
+    attribute :priority,      Integer
+    attribute :groups,        Array[Symbol]
+    attribute :graph_methods, Array[String]
```

Comment:
Ah, that's where that comes from! Thanks 👍 I was already really confused how that came about.

Reply:
Could you maybe help me with this. How would I do this without having these values saved into the derived dataset? What's the alternative to using an `attribute` for an `.ad` file?

Reply:
> Could you maybe help me with this. How would I do this  [...]

Happy to help, but I'm not sure why `graph_values` is an attribute in the first place. Can you explain what breaks if you *don't* declare it as an attribute?

Reply:
Well, the intention was to add it to the `Edge` and `Node` `.ad`'s to make sure that people are not able to blindly set every attribute for every `Node` and `Edge` in the `graph_values.yml`. It's just there for the [whitelisting](https://github.com/quintel/atlas/pull/106/files#diff-83d6809c3db5afe6d9b81c19b59b8cbfR12). 

I think people would abuse the system if I'd give them 100% freedom.

Reply:
Oh, stupid me. The attribute is graph_**methods** not graph_**values**. 😊 

Of course it makes perfect sense for that to be an attribute.

Perhaps all the graph values are being dumped to YAML [because of `to_h`](https://github.com/quintel/atlas/pull/106/files#diff-6b577c0b5b437252bc873fbf4b9a222eR29)? I kinda hope that's not the case because it would be silly for Atlas to persist keys from that hash (`graph_values`) if they're not declared as an attribute, but it wouldn't surprise me.

Reply:
I'm also a bit wondering now why I even need the `graph_values` in the `to_h` method at all. 🤔

Reply:
> I'm also a bit wondering now why I even need the graph_values in the to_h method at all.

That's a good question. :+1:

I can confirm that Atlas *is* using `to_hash` to determine which values are saved to the ".ad" file, and that is why `graph_values` are being persisted. Pull request coming soon. 😇